.github/workflows/ci.yml

```diff
@@ -64,27 +64,64 @@ jobs:
   windows_VS2017:
     name: "windows-vs2017"
     runs-on: windows-2016
-    timeout-minutes: 90
+    timeout-minutes: 120
+    env:
+      CLCACHE_DIR: ${{ GITHUB.WORKSPACE }}\clcache
+      CLCACHE_CL: C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\VC\Tools\MSVC\14.16.27023\bin\Hostx86\x86\cl_original.exe
     steps:
       - id: checkout
         uses: actions/checkout@v2
+      - id: cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CLCACHE_DIR }}
+          key: windows-vs2017-clcache-${{github.ref}}-${{github.sha}}
+          restore-keys: |
+            windows-vs2017-clcache-${{github.ref}}-
+            windows-vs2017-clcache-refs/heads/main-
       - name: Setup PATH
         uses: microsoft/setup-msbuild@v1.0.2
+      - name: Setup clcache
+        run: |
+          (New-Object System.Net.WebClient).DownloadFile('https://github.com/frerich/clcache/releases/download/v4.2.0/clcache-4.2.0.zip', "$pwd\clcache.zip")
```

Comment:
clcache is not maintained, so I'm not a fan of using it.

Reply:
I will give another go with sccache and see if I can make it work. Unfortunately I only found 3 options of caching for MSVC (2 of which are not maintained), but still from what I've found clcache seem to be the preferred one. There are some forks with active development, the author of clcache advises people to merge new required features to those repositories. Although I think for our use case the last release version may suffice if sccache cannot be used.

Reply:
Did you try setting `CMAKE_CXX_COMPILER=sccache cl`?

Reply:
I remember I tried `CMAKE_CXX_COMPILER_LAUNCHER` and it didn't work, maybe I tried this as well, but I'm not sure, I will give it a go.

Reply:
Unfortunately I could not make sccache work with cmake and msbuild. I tried setting cmake parameters like the `CMAKE_CXX_COMPILER` and `CMAKE_CXX_COMPILER_LAUNCHER` but sccache was not used. I also tried multiple methods mentioned in https://github.com/mozilla/sccache/issues/107, like setting the msbuild configuration parameters and replacing the cl.exe, but unfortunately non of them seem to work. Replacing the cl.exe with sccache and adjusting the PATH was the only method where I could see the cache calls in the sccache statistics, but all of them were marked as `Non-cacheable calls`.